src/Illuminate/View/ComponentAttributeBag.php

```diff
@@ -182,20 +199,46 @@ public function merge(array $attributeDefaults = [], $escape = true)
         }, $attributeDefaults);
 
         foreach ($this->attributes as $key => $value) {
-            if ($key !== 'class') {
-                $attributes[$key] = $value;
+            if ($this->attributeShouldBeImploded($key)) {
+                $attributes[$key] = implode($this->getAttributeImplodeGlueCharacter($key), array_unique(
+                    array_filter([$attributeDefaults[$key] ?? '', $value])
+                ));
 
                 continue;
             }
 
-            $attributes[$key] = implode(' ', array_unique(
-                array_filter([$attributeDefaults[$key] ?? '', $value])
-            ));
+            $attributes[$key] = $value;
         }
 
         return new static(array_merge($attributeDefaults, $attributes));
     }
 
+    /**
+     * Determine where or not to implode a merged attribute instead of overwriting it
+     *
+     * @param  string $key
+     * @return bool
+     */
+    protected function attributeShouldBeImploded($key)
+    {
+        return $key === 'class' || in_array($key, static::$implodedAttributes) || isset(static::$implodedAttributes[$key]);
+    }
+
+    /**
+     * Determine which character to use when imploding a merged attribute
+     *
+     * @param  string $key
```

Comment:
please follow laravel's phpdoc code style (not yet automatically enforced), everywhere